### ISSUE TITLE
Add DOS32A (DOS/32 Advanced DOS Extender) Compatibility

### ISSUE THREAD

**Issue by toniosj** [schellingb/dosbox-pure#562]
[https://www.dosbox.com/wiki/TOOLS:DOS32A](https://www.dosbox.com/wiki/TOOLS:DOS32A)

[https://github.com/yetmorecode/dos32a-ng](https://github.com/yetmorecode/dos32a-ng)

Games:
- Gal Pani (1995)
- Gal Pani II (1996)

**Comment by schellingb** [comment]
Is it not compatible with DOSBox Pure? How about regular DOSBox?

**Comment by PoloniumRain** [comment]
I quickly tried it with Screamer 1 and 2 and they both worked fine with DBP. But Screamer Rally with 3dfx acceleration wont run, it just goes back to the DOS prompt. Software mode works though.

All i did was put `DOS32ANG.EXE` in the games folder and rename it to `DOS4GW.EXE`. Might try benchmarking it if i get time, being as it's meant to "run faster and better".....  lol.

**Comment by schellingb** [comment]
Yes, I frequently read "better and faster" but then have no further information on why and how much faster. I'd be curious to see any benchmarking. I heard of DOSBox variants including either regular DOS32A or this new update but just including it is a bit meaningless if it still needs manual setting up per game. Having DOSBox automatically swap out DOS4GW for such a modern variant might be possible, but if it breaks games it is a no-go. And it sounds like it does break games... So the alternative would be to have a config option "Use DOS32A" and leave it to the user? But if we don't even know how useful it is, how would a regular user be able to decide. And then a technically minded user can just put DOS32A into the game on their own, without the core needing to do anything.

**Comment by Allanon41** [comment]
I've been using Dos32A renamed to Dos4gw (or just Dos32A as is) for a while now (even autoexec on my win98se build) and any "speed" improvements are negligible or unnoticeable. I was more interested in the added "stability" using higher memory >512 mb.

**Comment by PoloniumRain** [comment]
Forgot about testing this... So i've just compared `DOS32ANG.EXE` to a standard `DOS4GW.EXE` file in 9 different games, and there was no difference to frame rates, not even 1%, so i wont bother with benchmark tables, there's no point. 

It looks like the main benefit with `DOS32A.EXE` and `DOS32ANG.EXE` is that they can get certain games working with higher RAM capacities when an old version of `DOS4GW.EXE` has been included with the game. But i'd imagine that using a newer version of `DOS4GW.EXE` would also fix that problem...

It seems like claims of "faster and better" and better stability will simply be down to some games benefitting from the increased RAM.

I've always used the eXoDOS version of games, which will typically include a more recent version of `DOS4GW.EXE`, or the best specific version for each game, or some other memory patch/fix. So i can't remember ever running into memory related issues, even though i'll almost always set DBP to 512MB for most games, which is totally overkill.
